sky/resources.py

```diff
@@ -501,8 +501,9 @@ def _set_accelerators(
 
             # Canonicalize the accelerator names.
             accelerators = {
-                accelerator_registry.canonicalize_accelerator_name(acc):
-                acc_count for acc, acc_count in accelerators.items()
+                accelerator_registry.canonicalize_accelerator_name(
+                    acc, self._cloud): acc_count
```

Comment:
Minor: What happens if a user has 1 cloud only as indicated by `sky check` but `self._cloud` is not specified? I guess we can leave that to the future.

Reply:
That is a good point! We can consider inferring the cloud for the whole Resources when only one cloud is enabled. Let's leave it to the future as that may need more test. : )

Reply:
Actually, this might be debatable, as this may make an original working yaml fails, if a user add a new cloud.